src/filesystem/drive/inner.rs

```diff
@@ -95,15 +120,36 @@ impl InnerDrive {
 
             // Update Size:
             let new_children_size = node.ordered_child_pids().iter().fold(0, |acc, child_pid| {
+                // This should maybe actually float up an error as opposed to defaulting to 0 if a child can not be found using `try_fold`
+                // It implies a child on the current node can't be found in the filesystem.
+                // This should be done carefully though, maybe finishing the rest of its work before returning an error? Or maybe not?
                 let child_size = self.by_perm_id(child_pid).ok().map_or(0, Node::size);
                 acc + child_size
             });
-            let node_mut = self.by_id_mut(node_id).unwrap();
+            let node_mut = self
+                .by_id_mut_untracked(node_id)
+                .expect("We've already accessed this node immutably just above");
             match node_mut.data_mut().await {
                 NodeData::Directory { children_size, .. } => *children_size = new_children_size,
                 _ => {}
             }
-            // Update Cid:
+
+            // Update child CIDs
+            let child_pids = node_mut.data().ordered_child_pids();
+            let mut child_cids = Vec::new();
+            for pid in child_pids {
+                child_cids.push((pid, self.by_perm_id_mut_untracked(&pid)?.cid().await?))
+            }
+
+            let node_mut = self
+                .by_id_mut_untracked(node_id)
+                .expect("We've already accessed this node immutably just above");
```

Comment:
Do you need to grab a fresh handle on this? Isn't it still available?

Reply:
This is me dancing around the borrow checker. At line 141 I mutably borrow self  when I call `self.by_perm_id_mut_untracked`. By grabbibg a fresh handle down here I let the compiler drop the first borrow, at line 129, so it can do the borrow at 141. 

To avoid that I could potentially skip using the member functions to get mutable handles manually